src/core/machine-id-setup.c

```diff
@@ -190,7 +190,7 @@ static int generate_machine_id(char id[34], const char *root) {
         return 0;
 }
 
-int machine_id_setup(const char *root) {
+int machine_id_setup(const char *root, const char machine_id[34]) {
```

Comment:
Just for the consistent coding style, the rest of the machine-id code uses a 34 byte array for arguments. I assumed the original author wanted it that way, though I agree it is a bit cumbersome.
Happy to change as long as the inconsistency is tolerable.